src/models/propulsion/FGTurbine.cpp

```diff
@@ -503,6 +500,26 @@ bool FGTurbine::Load(FGFDMExec* exec, Element *el)
   MaxThrustLookup = GetPreFunction(property_prefix+"/AugThrust");
   InjectionLookup = GetPreFunction(property_prefix+"/Injection");
 
+  JSBSim::Element* tsfcElement = el->FindElement("tsfc");
+  if (tsfcElement) {
+    delete TSFC;
```

Comment:
It is customary to protect such calls with a test that checks that `TSFC` is not a `nullptr` because, with exceptions, you never know... And the check doesn't hurt anyway
```suggestion
    if (TSFC) {
      delete TSFC;
      TSFC = nullptr;
    }
```
At some point, we'll need to replace all the `new` and `delete` in this class with `std::unique_ptr<>`

Reply:
Definitely missed adding the deletes in the destructor. And yep would be better/easier with `std::unique_ptr<>`.

In terms of the `nullptr` check before the call to `delete`, it may be customary but also old-fashioned :wink:

https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null

Reply:
> In terms of the `nullptr` check before the call to `delete`, it may be customary but also old-fashioned :wink:

Oops ! You are absolutely right :smile: Forget the check.